src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/TargetCoverageUtils.java

```diff
@@ -0,0 +1,95 @@
+     * @return read depth per base per homolog
+     */
+    public static double estimateReadDepthDensityFromPoissonModel(final double[] readCounts,
+                                                                  final int[] targetLengths,
+                                                                  final double[] multiplicativeBiases,
+                                                                  final int[] ploidies,
+                                                                  final double[] copyRatio,
+                                                                  final int[] mask) {
+        Utils.nonNull(readCounts, "Read count array must be non-null");
+        Utils.nonNull(targetLengths, "Target length array must be non-null");
+        Utils.nonNull(multiplicativeBiases, "Multiplicative bias array must be non-null");
+        Utils.nonNull(ploidies, "Ploidy array must be non-null");
+        Utils.nonNull(copyRatio, "Copy ratio array must be non-null");
+        Utils.nonNull(mask, "Mask array must be non-null");
+        final int numTargets = readCounts.length;
+        if (numTargets == 0) {
+            throw new UserException.BadInput("At least one target is required");
+        }
+        if (targetLengths.length != numTargets) {
+            throw new UserException.BadInput("Target lengths array must have the same length as read counts array");
+        }
+        if (multiplicativeBiases.length != numTargets) {
+            throw new UserException.BadInput("Multiplicative bias array must have the same length as read counts array");
+        }
+        if (ploidies.length != numTargets) {
+            throw new UserException.BadInput("Plodies array must have the same length as read counts array");
+        }
+        if (copyRatio.length != numTargets) {
+            throw new UserException.BadInput("Copy ratio array must have the same length as read counts array");
+        }
+        if (mask.length != numTargets) {
+            throw new UserException.BadInput("Mask array must have the same length as read counts array");
+        }
+
+        /* calculate total multiplicative factor */
+        final double[] lambda = IntStream.range(0, numTargets)
+                .mapToDouble(i -> targetLengths[i] * multiplicativeBiases[i] * ploidies[i] * copyRatio[i])
+                .toArray();
+
+        /* mask out targets that have exactly zero Poisson factor to avoid division by zero */
+        final int[] updatedMask = mask.clone();
+        IntStream.range(0, numTargets)
+                .filter(i -> lambda[i] == 0.0)
+                .forEach(i -> { updatedMask[i] = 0; lambda[i] = 1; });
+        if (Arrays.stream(updatedMask).filter(m -> m != 0 && m != 1).count() > 0) {
+            throw new UserException.BadInput("Target mask may only have 0 and 1 values");
+        }
+        final long maskSum = Arrays.stream(updatedMask).mapToLong(m -> (long)m).sum();
+        if (maskSum == 0) {
```

Comment:
For consistency you could put the expression `final long maskSum = Arrays.stream(updatedMask).mapToLong(m -> (long)m).sum();` inside the if statement